mmpose/datasets/pipelines/mesh_transform.py

```diff
@@ -0,0 +1,370 @@
+
+    joints_3d_flipped = joints_3d.copy()
+    joints_3d_visible_flipped = joints_3d_visible.copy()
+
+    # Swap left-right parts
+    for left, right in flip_pairs:
+        joints_3d_flipped[left, :] = joints_3d[right, :]
+        joints_3d_flipped[right, :] = joints_3d[left, :]
+
+        joints_3d_visible_flipped[left, :] = joints_3d_visible[right, :]
+        joints_3d_visible_flipped[right, :] = joints_3d_visible[left, :]
+
+    # Flip horizontally
+    joints_3d_flipped[:, 0] = -joints_3d_flipped[:, 0]
+    joints_3d_flipped = joints_3d_flipped * joints_3d_visible_flipped
+
+    return joints_3d_flipped, joints_3d_visible_flipped
+
+
+@PIPELINES.register_module()
+class LoadIUVFromFile(object):
+    """Loading IUV image from file."""
+
+    def __init__(self, to_float32=False):
+        self.to_float32 = to_float32
+        self.color_type = 'color'
+        # channel relations: iuv->bgr
+        channel_order = 'bgr'
+        self.channel_order = channel_order
+
+    def __call__(self, results):
+        """Loading image from file."""
+        has_iuv = results['has_iuv']
+        use_iuv = results['ann_info']['use_IUV']
+        if has_iuv and use_iuv:
+            iuv_file = results['iuv_file']
+            iuv = mmcv.imread(iuv_file, self.color_type, self.channel_order)
+            if iuv is None:
+                raise ValueError('Fail to read {}'.format(iuv_file))
+        else:
+            has_iuv = 0
+            iuv = None
+
+        results['has_iuv'] = has_iuv
+        results['iuv'] = iuv
+        return results
+
+
+@PIPELINES.register_module()
+class IUVToTensor():
+    """Transform IUV image to Tensor.
+
+    Required key: 'iuv'. Modifies key: 'iuv'.
+
+    Args:
+        results (dict): contain all information about training.
+    """
+
+    def __call__(self, results):
+        iuv = F.to_tensor(results['iuv'])
+        iuv[0, :, :] = iuv[0, :, :] * 255
+        results['iuv'] = iuv
+        return results
+
+
+@PIPELINES.register_module()
+class MeshRandomChannelNoise():
+    """Data augmentation with random channel noise.
+
+    Required keys: 'img'
+    Modifies key: 'img'
+
+    Args:
+        noise_factor (float): Multiply each channel with
+         a factor between``[1-scale_factor, 1+scale_factor]``
+    """
+
+    def __init__(self, noise_factor=0.4):
+        self.noise_factor = noise_factor
+
+    def __call__(self, results):
+        """Perform data augmentation with random channel noise."""
+        img = results['img']
+
+        # Each channel is multiplied with a number
+        # in the area [1-self.noise_factor, 1+self.noise_factor]
+        pn = np.random.uniform(1 - self.noise_factor, 1 + self.noise_factor, 3)
+        img[:, :, 0] = np.minimum(255.0, np.maximum(0.0, img[:, :, 0] * pn[0]))
+        img[:, :, 1] = np.minimum(255.0, np.maximum(0.0, img[:, :, 1] * pn[1]))
+        img[:, :, 2] = np.minimum(255.0, np.maximum(0.0, img[:, :, 2] * pn[2]))
```

Comment:
```python
pn = np.random.uniform(1 - self.noise_factor, 1 + self.noise_factor, (1, 1, 3))
img = np.clip(img * pn, 0, 255)
```

Reply:
what's the dtype of img before this opeartion? is it uint8? if so, should you convert it back to uint8?

Reply:
The dtype of img is uint8, and after this operation, it's still uint8.

Reply:
okay, current version changes the array in-place